[DOC-373] Code location history (#24002)
dagster-io/dagster: docs/docs-beta/docs/dagster-plus/deployment/code-locations/code-location-history.md

---

> title: "Code location history"

Reviewer: This doesn't need to be merge-blocking, but I think we could get a more actionable/useful title here.

Ideally a user would be able to tell from the sidebar and title that this is where they'd go to rollback if they have a problem. wdyt, @cmpadden? I'm struggling to come up with something that isn't a mouthful.

Reviewer: But in that case, what does `deployment` refer to? Rolling back an entire full deployment? We use that term in a number of ways, unfortunately :|

Reviewer 2: Good point.

"image rollback", "version rollback", "Code location history and image rollback"

You're right that they're a mouthful, hah.

Author: I updated to "Code location history and rollbacks" -- it's a bit vague but at least it gives people who explicitly are looking for rollbacks a breadcrumb

displayed_sidebar: "dagsterPlus"
sidebar_position: 4
sidebar_label: "Code location history"
---

> 

Reviewer: ☝️ that

For prereqs, are there any Dagster+ permissions required to do anything in this guide?

> 

Author: you need editor permissions for rollbacks, so I'll add a note in that section

> 

Author: @cmpadden Struggling to think of what to put in the "what you'll learn" section that isn't just a duplicate of the two section titles haha -- any thoughts there?

> 

Reviewer 2: I think it's okay to be a little repetitive, I've added a basic "what you'll learn" section

# Code location history

Dagster+ automatically tracks metadata every time a code location is loaded. This can be used to understand when changes have been made, and what those changes were. In addition, this metadata can be used to quickly redeploy an older version.

## Viewing code location history

1. In the Dagster+ UI, navigate to the **Deployment** tab.

> 2. In the row associated with the code location you're interested in, click **View history** on the **Updated** column.

Reviewer: ```suggestion
2. In the row associated with the code location you're interested in, click **View history** in the **Updated** column.
```

Sorry for the nit - but it's inside the column

![Screenshot highlighting the "Updated" column for a code location)](/img/placeholder.svg)

This will bring up a modal showing a history of every time that code location has been loaded, and metadata associated with that load.

If you have connected Dagster+ to a GitHub or Gitlab repository, each row will have a link to the commit that was deployed at that point in time.

If a code location has been deployed multiple times with identical metadata, these rows will be collapsed together. You can expand them by deselecting **Collapse similar entries** in the top left corner of the modal.

> 

Reviewer 2: ```suggestion
```

> 

Reviewer 2: ```suggestion
```

Empty line

## Rolling back to a previous code location version

If you notice an issue with newly deployed code, or your code fails to deploy successfully, you can quickly roll back to a previously deployed image that's known to work properly.

1. In the Dagster+ UI, navigate to the **Deployment** tab.

> 2. In the row associated with the code location you're interested in, click **View history** on the **Updated** column.

Reviewer: ```suggestion
2. In the row associated with the code location you're interested in, click **View history** in the **Updated** column.
```

Same here

> 3. In the dropdown menu next to **View metadata**, select **Rollback to this version**.

Reviewer 2: ```suggestion
3. In the **Actions** column click the dropdown menu to the right of **View metadata**, select **Rollback to this version**.
```

I will say this two-in-one style of button is slightly unintuitive, at least when trying to describe it.

> ![Screenshot highlighting the "Updated" column for a code location](/img/placeholder.svg)

Reviewer 2: Are there any "Next steps" we'd want to include at the end?

Author: added!

> ![Screenshot highlighting the "Updated" column for a code location](/img/placeholder.svg)

Reviewer: Also, does anything happen after the user selects this? Is there confirmation in the UI or something?

Author: There is no confirmation dialog here, it just goes right ahead and does it
